wallet-web/components/bond/utils.ts

```diff
@@ -0,0 +1,121 @@
+import { printableBalanceToNative } from '@nymproject/nym-validator-client/dist/currency'
+import { Gateway, MixNode } from '@nymproject/nym-validator-client/dist/types'
+import bs58 from 'bs58'
+import semver from 'semver'
+import { basicRawCoinValueValidation } from '../../common/helpers'
+import { NodeType } from '../../common/node'
+import { BondingInformation } from './NodeBond'
+
+export const validateAmount = (rawValue: string, minimum: string): boolean => {
+  // tests basic coin value requirements, like no more than 6 decimal places, value lower than total supply, etc
+  if (!basicRawCoinValueValidation(rawValue)) {
+    return false
+  }
+
+  // this conversion seems really iffy but I'm not sure how to better approach it
+  let nativeValueString = printableBalanceToNative(rawValue)
+  let nativeValue = parseInt(nativeValueString)
+  console.log(nativeValue, minimum)
+  return nativeValue >= parseInt(minimum)
+}
+
+export const validateKey = (key: string): boolean => {
+  // it must be a valid base58 key
+  try {
+    const bytes = bs58.decode(key)
+    // of length 32
+    return bytes.length === 32
+  } catch {
+    return false
+  }
+}
+
+export const isValidHostname = (value) => {
+  if (typeof value !== 'string') return false
+
+  const validHostnameChars = /^[a-zA-Z0-9-.]{1,253}\.?$/g
+  if (!validHostnameChars.test(value)) {
+    return false
+  }
+
+  if (value.endsWith('.')) {
+    value = value.slice(0, value.length - 1)
```

Comment:
I'm not entirely sure I understand that, why are we removing the last character if it's a `.`?